### PR TITLE
fix(general): Correctly handle cli graphs in case we run with multiprocessing

### PR DESCRIPTION
**Pull request by bo156**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

[//]: # "
    # PR Title
    Be aware that we use the title to create changelog automatically and therefore only allow specific prefixes
    - break:    to indicate a breaking change, this supersedes any of the types
    - feat:     to indicate new features or checks
    - fix:      to indicate a bugfix or handling of edge cases of existing checks
    - docs:     to indicate an update to our documentation
    - chore:    to indicate adjustments to workflow files or dependency updates
    - platform: to indicate a change needed for the platform
    Additionally a scope is needs to be added to the prefix, which indicates the targeted framework, in doubt choose 'general'.
    #    
    Allowed prefixs:
    ansible|argo|arm|azure|bicep|bitbucket|circleci|cloudformation|dockerfile|github|gha|gitlab|helm|kubernetes|kustomize|openapi|sca|secrets|serverless|terraform|general|graph|terraform_plan|terraform_json
    #
    ex.
    feat(terraform): add CKV_AWS_123 to ensure that VPC Endpoint Service is configured for Manual Acceptance
"

## Description

When running for more than 1 framework we trigger a parallel run using `multiprocessing`, which relies on an update to the `runner` object which happens in another process. Instead, we now return it as well.

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my feature, policy, or fix is effective and works
- [ ] New and existing tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by YaaraVerner — APPROVED** [review]

File: checkov/common/runners/runner_registry.py [hunk 1]
```diff
@@ -134,7 +134,10 @@ def _parallel_run(runner: _BaseRunner) -> Report | list[Report]:
                     logging.error(f"Failed to create report for {runner.check_type} framework")
                     report = Report(check_type=runner.check_type)
 
-                return report
+                if runner.graph_manager:
+                    check_type_to_graph = {runner.check_type: runner.graph_manager.get_reader_endpoint()}
+                    return report, check_type_to_graph
+                return report, {}
```
Comment (gruebel) [inline checkov/common/runners/runner_registry.py:140]:
```suggestion
                if runner.graph_manager:
                    check_type_to_graph = {runner.check_type: runner.graph_manager.get_reader_endpoint()}
                    return report, runner.check_type, runner.graph_manager.get_reader_endpoint()
                return report, None, None
```
how about returning all as a normal tuple, then you can check if both are not `None` and update `full_check_type_to_graph[check_type] = reader_endpoint`

Comment (bo156) [reply checkov/common/runners/runner_registry.py]: fixed

File: checkov/common/runners/runner_registry.py [hunk 2]
```diff
@@ -161,7 +164,15 @@ def _parallel_run(runner: _BaseRunner) -> Report | list[Report]:
...
                     logging.log(level, f'The framework "{runner.check_type}" is part of the "{self.licensing_integration.get_subscription_for_runner(runner.check_type).name}" module, which is not enabled in the platform')
 
-            reports = [r for r in parallel_runner.run_function(func=_parallel_run, items=valid_runners, group_size=1) if r]
+            parallel_runner_results = parallel_runner.run_function(func=_parallel_run, items=valid_runners,
+                                                                   group_size=1)
+            reports: list[Report | list[Report]] = []
+            full_check_type_to_graph = {}
+            for report, check_type_to_graph in parallel_runner_results:
+                reports.append(report)
+                if check_type_to_graph != {}:
```
[the remark this answers is not in the document]
Comment (bo156) [reply checkov/common/runners/runner_registry.py]: now it is

**Review by gruebel — APPROVED** [review]
🍰

**Pull request merged**